sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -862,5 +866,94 @@ def test_incorrect_extension(self):
                                                  temp_filepath)
 
 
+SAMPLE_PIPELINES_TEST_DATA_DIR = os.path.join(
+    os.path.dirname(__file__), 'test_data')
+
+SPECIAL_TEST_PY_FILES = {'two_step_pipeline.py'}
+
+TEST_PY_FILES = {
+    file.split('.')[0]
+    for file in os.listdir(SAMPLE_PIPELINES_TEST_DATA_DIR)
```

Comment:
I actually prefer explicitly listing all the test files in code instead of `listdir`. Although it requires a bit overhead when adding/deleting a test sample, it gives the following benefits:
- When I search for the filename of test sample in code, I could find where/how it is referenced--a bit more readable and easy to follow IMO.
- Previously when I made some dramatic code changes to compiler, it often broke multiple samples during the development progress, and I could easily comment out the failed cases and address them one by one.

But I want to hear your opinion as well before making that change. Maybe there's a better dev workflow that I missed?
@connor-mccarthy

Reply:
Those are two good points and I've experienced the second directly. I agree -- let's list out all the test cases.